orbit/cmd/orbit/orbit.go

```diff
@@ -404,21 +405,42 @@ func main() {
 
 			g.Add(updateRunner.Execute, updateRunner.Interrupt)
 
-			osquerydLocalTarget, err := updater.Get("osqueryd")
-			if err != nil {
-				return fmt.Errorf("get osqueryd target: %w", err)
-			}
-			osquerydPath = osquerydLocalTarget.ExecPath
-			if c.Bool("fleet-desktop") {
-				fleetDesktopLocalTarget, err := updater.Get("desktop")
+			// if getting any of the targets fails, keep on
+			// retrying, the `updater.Get` method has built-in backoff functionality.
+			//
+			// NOTE: it used to be the case that we would return an
+			// error here, causing orbit to restart. This was
+			// changed to have control over how/when we want to
+			// retry to download the packages.
+			err = retrypkg.Do(func() error {
+				osquerydLocalTarget, err := updater.Get("osqueryd")
 				if err != nil {
-					return fmt.Errorf("get desktop target: %w", err)
+					log.Info().Err(err).Msg("get osqueryd target failed")
+					return fmt.Errorf("get osqueryd target: %w", err)
 				}
-				if runtime.GOOS == "darwin" {
-					desktopPath = fleetDesktopLocalTarget.DirPath
-				} else {
-					desktopPath = fleetDesktopLocalTarget.ExecPath
+				osquerydPath = osquerydLocalTarget.ExecPath
+				if c.Bool("fleet-desktop") {
+					fleetDesktopLocalTarget, err := updater.Get("desktop")
+					if err != nil {
+						log.Info().Err(err).Msg("get desktop target failed")
+						return fmt.Errorf("get desktop target: %w", err)
+					}
+					if runtime.GOOS == "darwin" {
+						desktopPath = fleetDesktopLocalTarget.DirPath
+					} else {
+						desktopPath = fleetDesktopLocalTarget.ExecPath
+					}
 				}
+
+				return nil
+			},
+				// retry every 5 minutes to not flood the logs,
+				// but actual pings to the remote server are
+				// handled by `updater.Get`
+				retrypkg.WithInterval(5*time.Minute),
```

Comment:
sanity checking: does this value makes sense?

Reply:
I don't follow this part `but actual pings to the remote server are handled by updater.Get`

Reply:
sorry, bad wording there. I mean that `updater.Get` is in charge of the backoff mechanism that downloads binaries from the TUF server.